packages/txservice/src/chainreader.ts

```diff
@@ -464,7 +464,12 @@ export class ChainReader {
     let gelatoEstimatedFee: BigNumber | undefined;
     if (this.config[chainIdForTokenPrice] && this.config[chainIdForTokenPrice].gelatoOracle) {
       const inputDecimals = await this.getDecimalsForAsset(chainId, assetId);
-      gelatoEstimatedFee = await this.calculateGelatoFee(chainIdForGasPrice, assetId, gasLimit.toNumber());
+      gelatoEstimatedFee = await this.calculateGelatoFee(
+        chainIdForGasPrice,
+        assetId,
+        gasLimit.toNumber(),
+        parseInt(gasEstimate),
```

Comment:
This might be too large to parseInt right?

Reply:
I just changed it for unary operator +